packages/api/lib/granules.js

```diff
@@ -86,30 +86,36 @@ const deleteGranuleAndFiles = async ({
   granulePgModel = new GranulePgModel(),
   granuleModelClient = new Granule(),
 }) => {
-  const files = await knex(filePgModel.tableName)
-    .where({ granule_cumulus_id: pgGranule.cumulus_id });
+  if (pgGranule === undefined) {
+    // Delete only the Dynamo Granule and S3 Files
+    await granuleModelClient.delete(dynamoGranule);
+  } else {
+    // Delete PG Granule, PG Files, Dynamo Granule, S3 Files
+    const files = await knex(filePgModel.tableName)
+      .where({ granule_cumulus_id: pgGranule.cumulus_id });
+
+    await knex.transaction(async (trx) => {
+      await pMap(
+        files,
+        (file) => {
+          filePgModel.delete(trx, { cumulus_id: file.cumulus_id });
+        }
+      );
+
+      await granulePgModel.delete(trx, pgGranule);
+      await granuleModelClient.delete(dynamoGranule);
+    });
```

Comment:
So basically if we don't have a PG granule, just do what we were doing previously and use the Dynamo model's `delete()` to delete the Dynamo record and the S3 files.

If we do have a PG Granule, do all the things. Delete the PG Granule, the Dynamo granule, the PG Files, and the S3 files.